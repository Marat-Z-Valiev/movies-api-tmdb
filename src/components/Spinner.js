import React from "react";
import styled from "styled-components";
import GridLoader from "react-spinners/GridLoader";

const SpinnerStyled = styled.div`
	div:nth-child(1) {
		margin: 0 auto;
	}
`;

const Spinner = () => {
	return (
		<>
			<SpinnerStyled>
				<GridLoader />
			</SpinnerStyled>
			;
		</>
	);
};

export default Spinner;
