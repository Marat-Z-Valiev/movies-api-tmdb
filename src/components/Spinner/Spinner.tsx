import React from "react";
import styled from "styled-components";
import fadeInAnimation from "../../animation/fadeInAnimation";
import GridLoader from "react-spinners/GridLoader";

const SpinnerStyled = styled.div`
	div:nth-child(1) {
		margin: 0 auto;
		${fadeInAnimation}
	}
`;

const Spinner = () => {
	return (
		<>
			<SpinnerStyled>
				<GridLoader />
			</SpinnerStyled>
			;
		</>
	);
};

export default Spinner;
