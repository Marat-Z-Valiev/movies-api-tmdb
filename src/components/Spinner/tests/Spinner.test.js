import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Spinner from "../Spinner";

configure({adapter: new Adapter()});

describe("Spinner", () => {
	it("should render correctly", () => {
		const wrapper = shallow(<Spinner />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
