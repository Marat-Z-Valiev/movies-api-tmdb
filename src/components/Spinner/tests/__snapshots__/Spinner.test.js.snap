// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render correctly 1`] = `
<Fragment>
  <Spinner__SpinnerStyled>
    <Loader
      color="#000000"
      css=""
      loading={true}
      margin={2}
      size={15}
    />
  </Spinner__SpinnerStyled>
  ;
</Fragment>
`;
